nvflare/fuel/hci/security.py

```diff
@@ -80,3 +80,41 @@ def get_certificate_common_name(cert: dict):
         for key, value in sub:
             if key == "commonName":
                 return value
+
+
+def get_certificate_roles(cert: dict):
+    """Gets the roles from the unstructuredName field of the provided certificate.
+
+    Args:
+        cert: certificate
+
+    Returns: roles of provided cert
+
+    """
+    if cert is None:
+        return None
+
+    for sub in cert.get("subject", ()):
+        for key, value in sub:
+            if key == "unstructuredName":
```

Comment:
should we use NameOID.ORGANIZATION_NAME to replace this string?

Reply:
roles are stored under Subject.unstructuredName in the form of "role1|role2|role3|...".  Because this function is to get the roles of peer cert, the key will be "unstructuredName."
From openssl:
Subject: CN = admin@nvidia.com, O = nvidia, unstructuredName = super

Reply:
In peer cert (ssl package), all keys are string.  The NameOID is only used in the cryptography package.

[peer cert](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket.getpeercert)
[NameOID](https://cryptography.io/en/latest/x509/reference/#cryptography.x509.oid.NameOID)